src/ServiceStack/HttpRequestExtensions.cs

```diff
@@ -317,6 +317,16 @@ public static int ToStatusCode(this Exception ex)
 
         public static string ToErrorCode(this Exception ex)
         {
+            if (HostContext.Config != null)
+            {
+                var exType = ex.GetType();
+                foreach (var entry in HostContext.Config.MapExceptionToErrorCode)
+                {
+                    if (entry.Key == exType)
```

Comment:
As error codes tend to be more specific than HTTP Status Codes I think subclasses should be explicitly customized.